### PR TITLE
Map arrows

### LINKED ISSUE #90: Create Edge Markers to point towards nearest story
A user should see arrows when no markers are in sight

### PR DESCRIPTION
**Pull request by aspensantos**
Capture

Closes #90

### CODE REVIEW

**Comment by AbhiPrasad** [comment]
We worked on https://github.com/uwblueprint/shoe-project/pull/143 to simplify frontend state a little bit.

From skimming over the PR, it doesn't seem too bad to rebase this PR on top of that, but curious to hear your thoughts. If it's a lot of trouble, we can merge this first, and then do the state stuff.

**Comment by aspensantos** [comment]
> We worked on #143 to simplify frontend state a little bit.
> 
> From skimming over the PR, it doesn't seem too bad to rebase this PR on top of that, but curious to hear your thoughts. If it's a lot of trouble, we can merge this first, and then do the state stuff.

Since this only uses the story state it should be okay to rebase, I will do that once the other PR is merged.

File: ui/src/components/Arrow.tsx
Comment (AbhiPrasad) [inline ui/src/components/Arrow.tsx]: Can we do string interpolation with template literals here? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

so:

```
<svg width='70' height='70' viewBox='-10 -10 80 80' fill='none' xmlns='http://www.w3.org/2000/svg' >
        <path transform='scale(0.3) translate(`${y}, ${x}) rotate(${rotation} blah blah
```

File: ui/src/components/EdgePinCluster.tsx
Comment (AbhiPrasad) [inline ui/src/components/EdgePinCluster.tsx]: Prefer arrow function in this case.

```suggestion
const createClusterCustomIcon = (edgeType: EdgeType) => {
```

File: ui/src/components/Arrow.tsx
Comment (AbhiPrasad) [inline ui/src/components/Arrow.tsx]: Can we make this func name a little more descriptive?

File: ui/src/components/EdgePin.tsx
Comment (AbhiPrasad) [inline ui/src/components/EdgePin.tsx]:
```suggestion
  return (
    <Marker position={position} icon={arrowIcon} />
  );
```

File: ui/src/components/EdgePin.tsx
Comment (AbhiPrasad) [inline ui/src/components/EdgePin.tsx:25]: Where are these numbers coming from? If they're just based on experiments, can we leave a comment explaining how we got there?

File: ui/src/components/EdgePinCluster.tsx
Comment (AbhiPrasad) [inline ui/src/components/EdgePinCluster.tsx]: Hmm can we just do this in a loop if everything stays the same but `EdgeType` and the array thats being mapped over?

Something like (sorry for bad formatting lol):

```ts
const arr = [{type: EdgeType.Right, cluster: rightCluster}, {type: EdgeType.Left, cluster: leftCluster} ... etc]
arr.map(({ type, cluster }) => {
  <MarkerClusterGroup
        showCoverageOnHover={false}
        spiderLegPolylineOptions={{ opacity: 0 }}
        iconCreateFunction={() => createClusterCustomIcon(type)}
      >
  {cluster.map((marker, key) => {
          return (
            <EdgePin
              key={key}
              position={marker.pos}
              angle={marker.angle}
              edgeType={marker.edgeType}
            />
          );
        })}
   </MarkerClusterGroup>
});
```

File: ui/src/components/EdgePins.tsx
Comment (AbhiPrasad) [inline ui/src/components/EdgePins.tsx:30]: Mind writing a comment how this works?

File: ui/src/components/EdgePins.tsx
Comment (AbhiPrasad) [inline ui/src/components/EdgePins.tsx]: Lets use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some here.

```suggestion
const isVisible = pinPositions.some((pos) => mapBounds.contains(L.latLng(pos.lat, pos.lng)));
```

File: ui/src/components/EdgePins.tsx
Comment (AbhiPrasad) [inline ui/src/components/EdgePins.tsx:136]: Nit: we can use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce to create an object. Ping me if you got questions about it.

Also fine with the current approach.

File: ui/src/components/EdgePins.tsx
Comment (AbhiPrasad) [inline ui/src/components/EdgePins.tsx]: Slice doesn't mutate the array, does this do what you want it to do?

File: ui/src/components/EdgePins.tsx
Comment (AbhiPrasad) [inline ui/src/components/EdgePins.tsx]: We can use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map here

File: ui/src/components/EdgePins.tsx
Comment (AbhiPrasad) [inline ui/src/components/EdgePins.tsx:158]: Can we pull this out into it's own function? Maybe even add some comments on what it is doing?

File: ui/src/components/EdgePins.tsx
Comment (AbhiPrasad) [inline ui/src/components/EdgePins.tsx]: We don't need this, we can just init the enum type and use it.

```ts
export enum EdgeType {
  Bottom = 0,
  Right,
  Top,
  Left,
}
```

then 
```ts
EdgeType.Bottom === 0 // true
```

**Review by AbhiPrasad — CHANGES_REQUESTED** [review]
Actual math wow. Nice work figuring this out!

Left some comments, mostly about javascript things. Ping me if you got questions about specifics.

**Review by hanlinc27 — COMMENTED** [review]
Can you update the redirect icon and styling before merge as well? 😃 @m-penny

File: ui/src/components/EdgePins.tsx
Comment (AbhiPrasad) [inline ui/src/components/EdgePins.tsx:143]: Isn't distances is a single element array because of `slice(0, 1)`?

File: ui/src/components/Arrow.tsx
Comment (AbhiPrasad) [reply ui/src/components/Arrow.tsx]: This still needed?

**Review by AbhiPrasad — APPROVED** [review]
I think we good to merge after CI is green (left 2 comments)

File: ui/src/components/Arrow.tsx
Comment (aspensantos) [reply ui/src/components/Arrow.tsx]: Yes but I updated it with the new styling

**Comment by aspensantos** [comment]
> Can you update the redirect icon and styling before merge as well? 😃 @m-penny

Yes, updated :)

File: ui/src/components/EdgePins.tsx
Comment (aspensantos) [reply ui/src/components/EdgePins.tsx:136]: Just left it got tired 🥴

File: ui/src/components/EdgePins.tsx
Comment (aspensantos) [reply ui/src/components/EdgePins.tsx:143]: Yeah this is because design changed it from the 3 closest to the closest one, so I made it so we could have more than one, but now it is just one. Not sure if I should leave it in case we want more in the future?

**Comment by dhillondeep** [comment]
These changes caused few bugs and slowdown on the map so, I rollbacked the changes and we can deploy this again when these issues are resolved

**Pull request merged**
